### DISCUSSION TITLE
Best way to model infiniband connections?

### DISCUSSION THREAD

**Discussion by msherman64** [netbox-community/netbox#8948] [Q&A]
I'm attempting to model a variety of Infiniband connections, but the data model doesn't seem a great fit.

Specifically, I have a HDR infiniband switch (50g per lane, 4x lanes per qsfp28 port), and it connects to servers either at 100G (HDR100, 2 lanes of 50G via breakout), or 200G (HDR, 4 lanes of 50G).

To fully describe a port, we'd need to specify both the datarate per-lane, and the number of lanes. This is partially addressed in https://github.com/netbox-community/netbox/issues/6303, as 4 lanes is an almost universal default, and HDR100 is specifically used in product descriptions as the 2 lane configuration is unusual.

I'm not sure why that issue was closed without comment.

An alternative would be to either drop speeds entirely from the Infiniband port descriptions, or set them to the "4-lane" values.

Any thoughts on a way forward? For now there's no way to properly specify the HDR100 ports, since they're 100g speed, but HDR protocol.

**Comment by krono** [comment]
I wrote that issue and was similarly dismayed at the commentless closure.
It seems that even trivial fixes are not redgarded at all if they do not "generate enought traffic" :(

**Reply by DanSheps** [reply]
> I wrote that issue and was similarly dismayed at the commentless closure. It seems that even trivial fixes are not redgarded at all if they do not "generate enought traffic" :(

There is only so much time in the day for us to review and work on issues, while I understand it might be disheartening to see a issue closed due to no-activity (which is the case here). 

While it was ultimately closed due to non-activity, there is another problem in that you are removing "information" from the database.  In this instance, it might be better to opt for adding additional interface types to add 2 lane, 3 lane and 4 lane HDR.

Since this was over a year ago, I would say go ahead and create a new FR to either remove the data rates or add additional interface types.  If you are willing to take care of that as well, feel free to volunteer as such in the issue.

**Comment by DanSheps** [comment]
> I'm attempting to model a variety of Infiniband connections, but the data model doesn't seem a great fit.
> 
> Specifically, I have a HDR infiniband switch (50g per lane, 4x lanes per qsfp28 port), and it connects to servers either at 100G (HDR100, 2 lanes of 50G via breakout), or 200G (HDR, 4 lanes of 50G).
> 
> To fully describe a port, we'd need to specify both the datarate per-lane, and the number of lanes. This is partially addressed in #6303, as 4 lanes is an almost universal default, and HDR100 is specifically used in product descriptions as the 2 lane configuration is unusual.
> 
> I'm not sure why that issue was closed without comment.

We track whether a issue is highly interesting for the community as a whole by both comments and "upvotes" (thumbs up).  This issue went stale due to not enough interest.

You are welcome to "re-open" a issue if there is additional information/more changes that come to light or sufficient time has passed.  For example, in this case you could "re-open" this to instead add additional data rates or if there is more you are willing to expand on how this should not include the data rates. 

Additionally, we will generally look at allowing something that might not garner much interest if someone is willing to own it and it is a relatively light change (such as removing data rates from the type or adding interface types/data rates or modifying the code in a minor, non-impacting manner).

**Reply by candlerb** [reply]
> You are welcome to "re-open" a issue

Just to clarify, do you mean opening a new issue for the same topic?  (It doesn't appear possible to re-open an existing closed/locked issue)

**Reply by DanSheps** [reply]
Yes, I mean open a new issue with for the same or similar.
